Class/Practise 5.2/include/MegaDataPool.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <mutex>
+#include "MegaData.hpp"
+
+class MegaDataPool {
+public:
+    static MegaDataPool& getInstance(size_t poolSize = 10) {
+        static MegaDataPool instance(poolSize);
+        return instance;
+    }
+
+    [[nodiscard]] std::shared_ptr<MegaData> acquire();
+    void release(size_t index);
+
+    [[nodiscard]] size_t size() const;
+    [[nodiscard]] size_t usedSize() const;
+
+    MegaDataPool(const MegaDataPool&) = delete;
```

Comment:
Minor: Just to follow [the rule of five](https://www.geeksforgeeks.org/rule-of-five-in-cpp/) we may need to delete move constructor here as well.